probe/consul.go

```
@@ -0,0 +1,131 @@
+			log.Printf("Consul query failed for %s (dc: %s, service: %s): %s", destinations[i].raw, destinations[i].datacenter, destinations[i].service, err)
+		}
+		endpointName := getEndpointFromConsul(destinations[i].service, *cfg.EndpointSuffix, endpointEntries)
+		minioClient, err := newMinioClientFromEndpoint(endpointName, *cfg.AccessKey, *cfg.SecretKey)
+		if err != nil {
+			log.Printf("Could not create minio client for %s (dc: %s, service: %s) : %s", destinations[i].raw, destinations[i].datacenter, destinations[i].service, err)
+		}
+		s3endpoints = append(s3endpoints, s3endpoint{name: endpointName, s3Client: minioClient})
+		log.Printf("Added gateway destination: %s", endpointName)
+	}
+	return s3endpoints
+}
+
+type destination struct {
+	datacenter string
+	service    string
+	raw        string
+}
+
+func extractDestinations(serviceEntries []*consul_api.ServiceEntry) (destinations []destination) {
+	ok := false
+	rawDestinations := ""
+	for i := range serviceEntries {
+		rawDestinations, ok = serviceEntries[i].Service.Meta["gateway_destinations"]
+		if ok {
+			break
+		}
+	}
+
+	log.Printf("Processing gateway destinations: %s", rawDestinations)
+	rawDestinationList := strings.Split(rawDestinations, ";")
+	re := regexp.MustCompile("^(.*):(.*)$")
+
+	for i := range rawDestinationList {
+		match := re.FindStringSubmatch(rawDestinationList[i])
+		if len(match) < 2 {
+			continue
+		}
+		destinations = append(destinations, destination{raw: match[0], datacenter: match[1], service: match[2]})
+	}
+	return destinations
+}
+
+// getServicePort return the first port found in the service or 80
+func getServicePort(serviceEntries []*consul_api.ServiceEntry) int {
+	port := 80
+	for i := range serviceEntries {
+		port = serviceEntries[i].Service.Port
+		break
+	}
+	return port
+}
+
+func getDatacenter(serviceEntries []*consul_api.ServiceEntry) string {
+	dc := ""
+	for i := range serviceEntries {
+		dc = serviceEntries[i].Node.Datacenter
+		break
+	}
+	return dc
```

should return an error if ""